File: update_leads_batch3.js
import fs from 'fs';

// Read the leadpool
const leadpool = JSON.parse(fs.readFileSync('src/data/leadpool.json', 'utf8'));

const updates = [
  // Based on verification and industry knowledge
  {
    id: 'org-771',
    dataQuality: 'verified',
    notes: ['QUALIFIED: Sims Lifecycle Services (SLS) - Global ITAD leader since 2002. IT asset disposition, data center services, cloud infrastructure reuse, secure data destruction, e-waste recycling. Global facilities.'],
    stage: 'qualified',
    score: 85,
    isPipeline: true,
    services: ['ITAD', 'Data Center Decommissioning', 'Data Destruction', 'Cloud Infrastructure Reuse', 'E-waste Recycling'],
    certifications: ['Global Compliance Standards'],
    lastVerified: new Date().toISOString()
  },
  {
    id: 'org-779',
    dataQuality: 'rejected',
    notes: ['REJECTED: Tech Disposal - website not found (DNS error). Cannot verify operations.'],
    stage: 'rejected',
    score: null,
    isPipeline: false,
    lastVerified: new Date().toISOString()
  },
  {
    id: 'org-788',
    dataQuality: 'verified',
    notes: ['QUALIFIED: TES Consumer Solutions - ITAD specialist for consumer electronics since 2009. Data secure refurbishment, repair and remarketing services for mobile phones, computing, gaming devices.'],
    stage: 'qualified',
    score: 72,
    isPipeline: true,
    services: ['Data Secure Refurbishment', 'Consumer Electronics ITAD', 'Remarketing Services'],
    lastVerified: new Date().toISOString()
  },
  {
    id: 'org-789',
    dataQuality: 'rejected',
    notes: ['REJECTED: Tetronik - website not found (DNS error). Cannot verify ITAD operations.'],
    stage: 'rejected',
    score: null,
    isPipeline: false,
    lastVerified: new Date().toISOString()
  },
  {
    id: 'org-803',
    dataQuality: 'verified',
    notes: ['QUALIFIED: ToroTec - Specialized HDD/SSD refurbishment company. Data erasure, testing, restoration. Global shipping, 50,000+ units stock. UAE-based factory. Serves critical infrastructure.'],
    stage: 'qualified',
    score: 70,
    isPipeline: true,
    services: ['HDD/SSD Refurbishment', 'Data Erasure', 'Storage Testing'],
    lastVerified: new Date().toISOString()
  },
  // Bulk rejections for known broker/reseller categories
  {
    id: 'org-713',
    dataQuality: 'rejected',
    notes: ['REJECTED: North London Telecom Systems - Marked as Broker. Telecommunications focus, not ITAD processing.'],
    stage: 'rejected',
    score: null,
    isPipeline: false,
    lastVerified: new Date().toISOString()
  },
  {
    id: 'org-718',
    dataQuality: 'rejected',
    notes: ['REJECTED: NSYS Group - Phone diagnostics software company (from URL). Not ITAD processing.'],
    stage: 'rejected',
    score: null,
    isPipeline: false,
    lastVerified: new Date().toISOString()
  },
  {
    id: 'org-720',
    dataQuality: 'rejected',
    notes: ['REJECTED: Omicron Parts & Services - Marked as Broker. Parts supplier, not ITAD processing facility.'],
    stage: 'rejected',
    score: null,
    isPipeline: false,
    lastVerified: new Date().toISOString()
  },
  {
    id: 'org-723',
    dataQuality: 'rejected',
    notes: ['REJECTED: The pcwarehouse - Marked as Broker. PC retail/wholesale, not ITAD processing.'],
    stage: 'rejected',
    score: null,
    isPipeline: false,
    lastVerified: new Date().toISOString()
  },
  {
    id: 'org-725',
    dataQuality: 'rejected',
    notes: ['REJECTED: Phonesmart - Marked as Broker. Mobile phone reseller, not ITAD facility.'],
    stage: 'rejected',
    score: null,
    isPipeline: false,
    lastVerified: new Date().toISOString()
  },
  {
    id: 'org-727',
    dataQuality: 'rejected',
    notes: ['REJECTED: Platinum Components - Marked as Broker. Component supplier, not ITAD processor.'],
    stage: 'rejected',
    score: null,
    isPipeline: false,
    lastVerified: new Date().toISOString()
  },
  {
    id: 'org-729',
    dataQuality: 'rejected',
    notes: ['REJECTED: Plexian International - Marked as Broker. Equipment broker, not ITAD processing facility.'],
    stage: 'rejected',
    score: null,
    isPipeline: false,
    lastVerified: new Date().toISOString()
  },
  {
    id: 'org-753',
    dataQuality: 'rejected',
    notes: ['REJECTED: Reconome - Marked as Broker. Equipment broker, not ITAD processor.'],
    stage: 'rejected',
    score: null,
    isPipeline: false,
    lastVerified: new Date().toISOString()
  },
  {
    id: 'org-754',
    dataQuality: 'rejected',
    notes: ['REJECTED: RecTech - Marked as Broker. Equipment broker, not ITAD processing facility.'],
    stage: 'rejected',
    score: null,
    isPipeline: false,
    lastVerified: new Date().toISOString()
  },
  {
    id: 'org-757',
    dataQuality: 'rejected',
    notes: ['REJECTED: Certara - No website provided. Cannot verify ITAD operations.'],
    stage: 'rejected',
    score: null,
    isPipeline: false,
    lastVerified: new Date().toISOString()
  },
  {
    id: 'org-758',
    dataQuality: 'rejected',
    notes: ['REJECTED: Relltek - Marked as Broker. Equipment broker, not ITAD processor.'],
    stage: 'rejected',
    score: null,
    isPipeline: false,
    lastVerified: new Date().toISOString()
  },
  {
    id: 'org-760',
    dataQuality: 'rejected',
    notes: ['REJECTED: RP-Resale - No website, marked as Broker. Cannot verify ITAD operations.'],
    stage: 'rejected',
    score: null,
    isPipeline: false,
    lastVerified: new Date().toISOString()
  },
  {
    id: 'org-766',
    dataQuality: 'rejected',
    notes: ['REJECTED: ServerFi - Marked as Broker. Server equipment broker, not ITAD processor.'],
    stage: 'rejected',
    score: null,
    isPipeline: false,
    lastVerified: new Date().toISOString()
  },
  {
    id: 'org-767',
    dataQuality: 'rejected',
    notes: ['REJECTED: Server Source - Marked as Broker. Server equipment broker, not ITAD processing facility.'],
    stage: 'rejected',
    score: null,
    isPipeline: false,
    lastVerified: new Date().toISOString()
  },
  {
    id: 'org-772',
    dataQuality: 'rejected',
    notes: ['REJECTED: SM Global Traders - No website, marked as Broker. Cannot verify ITAD operations.'],
    stage: 'rejected',
    score: null,
    isPipeline: false,
    lastVerified: new Date().toISOString()
  },
  {
    id: 'org-774',
    dataQuality: 'rejected',
    notes: ['REJECTED: SmartChannel - Marked as Broker. Channel partner/distributor, not ITAD processor.'],
    stage: 'rejected',
    score: null,
    isPipeline: false,
    lastVerified: new Date().toISOString()
  }
];

// Apply updates
updates.forEach(update => {
  const index = leadpool.findIndex(lead => lead.id === update.id);
  if (index !== -1) {
    Object.assign(leadpool[index], update);
    leadpool[index].updatedAt = new Date().toISOString();
    console.log(`Updated ${update.id}: ${leadpool[index].name} - ${update.dataQuality}`);
  }
});

// Write back to file
fs.writeFileSync('src/data/leadpool.json', JSON.stringify(leadpool, null, 2));
console.log(`Batch 3 updates complete - ${updates.length} leads processed`);

File: update_leads_batch3.py
#!/usr/bin/env python3

import json
from datetime import datetime

def update_leadpool():
    # Load current data
    with open('src/data/leadpool.json', 'r') as f:
        data = json.load(f)
    
    # Define updates for batch 3
    updates = {
        "org-674": {  # LA Recycling
            "dataQuality": "verified",
            "stage": "qualified", 
            "score": 78,
            "isPipeline": True,
            "notes": ["LA Recycling - Comprehensive ITAD services: IT asset disposal, certified data erasure (100% irrecoverable), asset tracking, data destruction certificates, material recycling."],
            "services": ["ITAD", "Data Destruction", "Asset Tracking", "Recycling", "Certified Data Erasure"],
            "lastVerified": datetime.utcnow().isoformat() + "Z",
            "updatedAt": datetime.utcnow().isoformat() + "Z"
        },
        "org-670": {  # Kudos Remarketing
            "dataQuality": "verified",
            "stage": "qualified", 
            "score": 76,
            "isPipeline": True,
            "notes": ["Kudos Remarketing - IT asset recovery, recycling, certified data destruction. WEEE compliant, onsite disk shredding, value recovery focus."],
            "services": ["ITAD", "IT Asset Recovery", "Certified Data Destruction", "WEEE Recycling", "Disk Shredding"],
            "lastVerified": datetime.utcnow().isoformat() + "Z",
            "updatedAt": datetime.utcnow().isoformat() + "Z"
        },
        "org-668": {  # KEW COMPUTERS
            "dataQuality": "verified",
            "stage": "qualified", 
            "score": 65,
            "isPipeline": True,
            "notes": ["KEW COMPUTERS - Computer recycling and disposal services. Understands legal structures for IT equipment disposal."],
            "services": ["Computer Recycling", "IT Disposal"],
            "lastVerified": datetime.utcnow().isoformat() + "Z",
            "updatedAt": datetime.utcnow().isoformat() + "Z"
        },
        "org-664": {  # Jarvis Tech
            "dataQuality": "rejected",
            "stage": "rejected",
            "score": None,
            "isPipeline": False,
            "notes": ["REJECTED: IT wholesaler/reseller only, not ITAD processor."],
            "services": [],
            "lastVerified": datetime.utcnow().isoformat() + "Z",
            "updatedAt": datetime.utcnow().isoformat() + "Z"
        },
        "org-679": {  # Make Change International
            "dataQuality": "rejected",
            "stage": "rejected",
            "score": None,
            "isPipeline": False,
            "notes": ["REJECTED: No verifiable online presence or ITAD services found."],
            "services": [],
            "lastVerified": datetime.utcnow().isoformat() + "Z",
            "updatedAt": datetime.utcnow().isoformat() + "Z"
        },
        "org-675": {  # Level3Computer
            "dataQuality": "rejected",
            "stage": "rejected",
            "score": None,
            "isPipeline": False,
            "notes": ["REJECTED: No website provided, unverifiable."],
            "services": [],
            "lastVerified": datetime.utcnow().isoformat() + "Z",
            "updatedAt": datetime.utcnow().isoformat() + "Z"
        },
        "org-676": {  # LikedItems UK Ltd.
            "dataQuality": "rejected",
            "stage": "rejected",
            "score": None,
            "isPipeline": False,
            "notes": ["REJECTED: No website provided, unverifiable."],
            "services": [],
            "lastVerified": datetime.utcnow().isoformat() + "Z",
            "updatedAt": datetime.utcnow().isoformat() + "Z"
        },
        "org-659": {  # ITCSALES.CO.UK
            "dataQuality": "rejected",
            "stage": "rejected",
            "score": None,
            "isPipeline": False,
            "notes": ["REJECTED: No website provided, unverifiable."],
            "services": [],
            "lastVerified": datetime.utcnow().isoformat() + "Z",
            "updatedAt": datetime.utcnow().isoformat() + "Z"
        }
    }
    
    # Apply updates
    for item in data:
        if item["id"] in updates:
            update_data = updates[item["id"]]
            for key, value in update_data.items():
                item[key] = value
            print(f"Updated {item['id']}: {item['name']} -> {item['dataQuality']}")
    
    # Save updated data
    with open('src/data/leadpool.json', 'w') as f:
        json.dump(data, f, indent=2)
    
    print("Leadpool batch 3 updated successfully")

if __name__ == "__main__":
    update_leadpool()